File: app/src/main/java/com/prey/actions/geofences/GeofenceDataSource.java
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2015 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.actions.geofences;

import android.content.Context;

import com.prey.PreyLogger;

import java.util.List;
import java.util.Map;

public class GeofenceDataSource {

    private GeofenceOpenHelper dbHelper;

    public GeofenceDataSource(Context context) {
        dbHelper = new GeofenceOpenHelper(context);
    }

    public void createGeofence(GeofenceDto geofence) {
        try {
            dbHelper.insertGeofence(geofence);
        } catch (Exception e) {;
            try {
                dbHelper.updateGeofence(geofence);
            } catch (Exception e1) {
                PreyLogger.e("error db update:" + e1.getMessage(), e1);
            }
        }
    }

    public void deleteGeofence(String id) {
        dbHelper.deleteGeofence(id);
    }

    public List<GeofenceDto> getAllGeofences() {
        return dbHelper.getAllGeofences();
    }

    public GeofenceDto getGeofences(String id) {
        return dbHelper.getGeofence(id);
    }

    public void deleteAllGeofence() {
        dbHelper.deleteAllGeofence();
    }

}

File: app/src/main/java/com/prey/actions/geofences/kotlin/GeofenceDataSource.kt
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2025 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.actions.geofences.kotlin

import android.content.Context
import com.prey.kotlin.PreyLogger

class GeofenceDataSource(context: Context?) {
    private val dbHelper = GeofenceOpenHelper(context)

    fun createGeofence(geofence: GeofenceDto) {
        try {
            dbHelper.insertGeofence(geofence)
        } catch (e: Exception) {
            try {
                dbHelper.updateGeofence(geofence)
            } catch (e1: Exception) {
                PreyLogger.e("GEO error db update:" + e1.message, e1)
            }
        }
    }

    fun deleteGeofence(id: String) {
        dbHelper.deleteGeofence(id)
    }

    val allGeofences: List<GeofenceDto>
        get() = dbHelper.allGeofences

    fun getGeofences(id: String): GeofenceDto? {
        return dbHelper.getGeofence(id)
    }

    fun deleteAllGeofence() {
        dbHelper.deleteAllGeofence()
    }

    fun updateGeofenceType(id: String, type: String?) {
        dbHelper.updateGeofenceType(id, type)
    }

    fun updateGeofence(geo: GeofenceDto) {
        dbHelper.updateGeofence(geo)
    }
}